pkg/isoeditor/rhcos.go

```diff
@@ -81,6 +98,38 @@ func (e *rhcosEditor) CreateMinimalISOTemplate(fullISOPath, rootFSURL, arch, min
 	return nil
 }
 
+func createNmstateRamDisk(arch, nmstatectlPath, ramDiskPath string) error {
+	// Ensure that the runtime CPU arch equals to the requested arch
+	if normalizeCPUArchitecture(runtime.GOARCH) != arch {
+		return nil
```

Comment:
I think this is a bit unexpected.

I'd rather `createNmstateRamDisk` fail if it doesn't ... create the nmstate ram disk :shrug: 

I'd say do this arch check in the caller and either don't call this function at all or error when the architecture doesn't match.